From 569ac1066cd3046b8ac899153df9f07908d45145 Mon Sep 17 00:00:00 2001
From: Mark Asselstine <mark.asselstine@windriver.com>
Date: Fri, 21 Jun 2013 11:16:00 -0400
Subject: [PATCH] openvswitch: add target perl handling

Allow the build to specify a path for the perl instead of reusing
the PERL variable which can lead to inconsistencies if we are cross
compiling. The TARGET_PERL variable will be used for script
substitutions to ensure the scripts will be able to properly execute
if the target system has a different path for the perl.

Signed-off-by: Mark Asselstine <mark.asselstine@windriver.com>
---
 Makefile.am                  | 1 +
 configure.ac                 | 7 +++++++
 utilities/ovs-parse-leaks.in | 2 +-
 3 files changed, 9 insertions(+), 1 deletion(-)

Index: openvswitch-2.0.0/Makefile.am
===================================================================
--- openvswitch-2.0.0.orig/Makefile.am
+++ openvswitch-2.0.0/Makefile.am
@@ -114,6 +114,7 @@
                 -e 's,[@]LOGDIR[@],$(LOGDIR),g' \
                 -e 's,[@]DBDIR[@],$(DBDIR),g' \
                 -e 's,[@]PERL[@],$(PERL),g' \
+                -e 's,[@]TARGET_PERL[@],$(TARGET_PERL),g' \
                 -e 's,[@]PYTHON[@],$(PYTHON),g' \
                 -e 's,[@]TARGET_PYTHON[@],$(TARGET_PYTHON),g' \
                 -e 's,[@]RUNDIR[@],$(RUNDIR),g' \
Index: openvswitch-2.0.0/configure.ac
===================================================================
--- openvswitch-2.0.0.orig/configure.ac
+++ openvswitch-2.0.0/configure.ac
@@ -115,6 +115,13 @@
 AC_SUBST(KARCH)
 OVS_CHECK_LINUX
 
+if test "$TARGET_PERL"; then
+    TARGET_PERL=$TARGET_PERL
+else
+    TARGET_PERL=$PERL
+fi
+AC_SUBST(TARGET_PERL)
+
 if test "$TARGET_PYTHON"; then
     TARGET_PYTHON=$TARGET_PYTHON
 else
